src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs

```diff
@@ -401,6 +406,130 @@ private static DateTime CreateDateTime(double ticks)
+                return null;
+            if (parmType.IsEnum)
+                return Runtime.EnumFromExportContract(parmType, obj);
+            else
+                return null;
+        }
+
+        private static Enum EnumFromExportContract(Type enumType, object value)
+        {
+            if (!enumType.IsEnum)
+            {
+                throw new ArgumentException("Type provided must be an Enum.", nameof(enumType));
+            }
+
+            if (value is string)
+            {
+                FieldInfo[] fields = enumType.GetFields();
+                foreach (FieldInfo fi in fields)
+                {
+                    // Do not process special names
+                    if (fi.IsSpecialName)
+                        continue;
+
+                    ExportAttribute[] attributes =
+                        (ExportAttribute[])fi.GetCustomAttributes(typeof(ExportAttribute), false);
+
+                    ConvertEnum enumConversionType = ConvertEnum.Default;
+
+                    object? contractName = null;
+
+                    if (attributes != null && attributes.Length > 0)
+                    {
+                        enumConversionType = attributes[0].EnumValue;
+                        if (enumConversionType != ConvertEnum.Numeric)
+                            contractName = attributes[0].ContractName;
+                    }
+
+                    contractName = contractName ?? fi.Name;
+                    switch (enumConversionType)
+                    {
+                        case ConvertEnum.ToLower:
+                            contractName = contractName!.ToString()!.ToLower();
+                            break;
+                        case ConvertEnum.ToUpper:
+                            contractName = contractName!.ToString()!.ToUpper();
+                            break;
+                        case ConvertEnum.Numeric:
+                            if (Enum.TryParse(enumType, value!.ToString() ?? string.Empty, out object? enumAsNumeric))
+                            {
+                                if (enumAsNumeric is Enum valueAsEnum)
+                                    return (Enum)valueAsEnum;
+                            }
+                            break;
+                        default:
+                            contractName = contractName!.ToString();
+                            break;
+                    }
+
+                    if (contractName!.ToString() == value.ToString())
+                    {
+                        return (Enum)Enum.Parse(enumType, fi.Name);
+                    }
+                }
+
+                throw new ArgumentException($"Value is a name, but not one of the named constants defined for the enum of type: {enumType}.", nameof(value));
+            }
+            else
+            {
+                return (Enum)Enum.ToObject(enumType, value);
+            }
+        }
+
+        private static object EnumToExportContract(Enum value)
```

Comment:
Why to return an object type if we always need a string?

Reply:
It can also return a primitive value as well.

Reply:
Sure but why should it return as object if we immediate convert it to string?